lib/livebook/teams/app_deployment.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Livebook.Teams.AppDeployment do
+  use Ecto.Schema
+  alias Livebook.FileSystem
+
+  @file_extension ".zip"
+
+  @type t :: %__MODULE__{
+          id: String.t() | nil,
+          filename: String.t() | nil,
+          slug: String.t() | nil,
+          sha: String.t() | nil,
+          title: String.t() | nil,
+          deployment_group_id: String.t() | nil,
+          file: {:url, String.t()} | {:content, binary()} | nil,
+          deployed_by: String.t() | nil,
+          deployed_at: NaiveDateTime.t() | nil
+        }
+
+  @primary_key {:id, :string, autogenerate: false}
+  embedded_schema do
+    field :filename, :string
+    field :slug, :string
+    field :sha, :string
+    field :title, :string
+    field :deployment_group_id, :string
+    field :file, :string
+    field :deployed_by, :string
+
+    timestamps(updated_at: nil, inserted_at: :deployed_at)
+  end
+
+  @doc """
+  Creates a new app deployment from notebook.
+  """
+  @spec new(Livebook.Notebook.t(), String.t(), Livebook.FileSystem.File.t()) ::
+          {:ok, t()} | {:warning, list(String.t())} | {:error, FileSystem.error()}
+  def new(notebook, filename, files_dir) do
+    filename = ensure_notebook_extension(filename)
```

Comment:
In this case the filename can be arbitrary and we shouldn't make any assumptions about it either, so I think we can simplify and just make it `notebook.livemd` :)